webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -46,6 +50,13 @@ export const apiRequest = async (
       response: response.data,
     });
 
+    updatedData &&
+      dispatch({
+        type: UPDATE,
+        key,
+        payload: updatedData,
```

Comment:
Can we pass here the `prevState` into the `updateData` as a callback?

the consumer can use it similiarly to React `useState(prevState => ({ ...prevState, ...changedState}))`

so I guess the consumer can do something like:
```js
post({
    key: MY_SPECIAL_KEY,
    url,
    params: { id }
    updateData: prevState => ({
        ...prevState,
        results: prevState.results.map(res => res.id === id ? changedResults : res)})
})
```

I know that you used `deep` update to redux store, but maybe as a callback user have more power and understanding of how this works.

Reply:
this is indeed better, then just requiring the full updated data blob, but I still believe we should be able to update the store from the response without the need of updateData.

Reply:
@ezr-ondrej if I understand correctly you mean triggering a `post(key, url, params)`, and right after that an `update(key, payload)` ?

Reply:
👍 for this callback idea @laviro
@ezr-ondrej, the `updateData` callback is optional, if the consumer won't add it to the api request, the response will be updated in the store according to the corresponding key:
```js
    const response = await getApiResponse({ type, url, headers, params });

    dispatch({
      type: SUCCESS,
      key,
      payload: modifiedPayload,
      response: response.data, // adds the data to the store
                              //  if the API returns the altered resources,
                             //  no need to pass the updatedData callback
    });

// optional by the consumer, should be used when api requests dosen't return all the altered resources
    updatedData && 
      dispatch({
        type: UPDATE,
        key,
        payload: updatedData(getState().key.response), // updatedData is a callback with prevState as argument
      });
```

This modification is for keeping the store once source of truth without duplicate the data